src/test/java/com/teragrep/pth10/DefaultTimeFormatTest.java

```diff
@@ -61,6 +61,24 @@ void enforceTimeZone() {
         TimeZone.setDefault(TimeZone.getTimeZone("Europe/Helsinki"));
     }
 
+    @Test
+    void ISO8601_fractions_latest_Test() {
+        String time = "2023-12-18T12:40:53.001+03:00"; // different timeformat than default (default would be +02:00)
+        long expected = 1702892453L + 1L;
+        long actual = new DefaultTimeFormat(true).getEpoch(time);
+
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void ISO8601_fractions_earliest_Test() {
+        String time = "2023-12-18T12:40:53.001+03:00"; // different timeformat than default (default would be +02:00)
+        long expected = 1702892453L;
+        long actual = new DefaultTimeFormat(false).getEpoch(time);
+
+        Assertions.assertEquals(expected, actual);
+    }
+
```

Comment:
Could you also add a test that a unparseable time throws an exception with a expected message?

Reply:
added